packages/kbn-flot-charts/lib/jquery_flot.js

```diff
@@ -1,3168 +1,3777 @@
+   *
+   * The idea is that that the width, height, and general 'identity' of the
+   * text is constant no matter where it is placed; the placements are a
+   * secondary property.
+   *
+   * Canvas maintains a cache of recently-used text info objects; getTextInfo
+   * either returns the cached element or creates a new entry.
+   *
+   * @param {string} layer A string of space-separated CSS classes uniquely
+   *     identifying the layer containing this text.
+   * @param {string} text Text string to retrieve info for.
+   * @param {(string|object)=} font Either a string of space-separated CSS
+   *     classes or a font-spec object, defining the text's font and style.
+   * @param {number=} angle Angle at which to rotate the text, in degrees.
+   * @param {number=} width Maximum width of the text before it wraps.
+   * @return {object} a text info object.
+   */
+  Canvas.prototype.getTextInfo = function (layer, text, font, angle, width) {
+    let textStyle;
+    let layerCache;
+    let styleCache;
+    let angleCache;
+    let info;
+
+    text = '' + text; // Cast to string in case we have a number or such
+    angle = (360 + (angle || 0)) % 360; // Normalize the angle to 0...359
+
+    // If the font is a font-spec object, generate a CSS font definition
+
+    if (typeof font === 'object') {
+      textStyle =
+        font.style +
+        ' ' +
+        font.variant +
+        ' ' +
+        font.weight +
+        ' ' +
+        font.size +
+        'px/' +
+        font.lineHeight +
+        'px ' +
+        font.family;
+    } else {
+      textStyle = font;
+    }
 
-		this.textContainer = null;
-		this.text = {};
+    // Retrieve or create the caches for the text's layer, style, and angle
 
-		// Cache of text fragments and metrics, so we can avoid expensively
-		// re-calculating them when the plot is re-rendered in a loop.
+    layerCache = this._textCache[layer];
+    if (layerCache == null) {
+      layerCache = this._textCache[layer] = {};
+    }
 
-		this._textCache = {};
-	}
+    styleCache = layerCache[textStyle];
+    if (styleCache == null) {
+      styleCache = layerCache[textStyle] = {};
+    }
 
-	// Resizes the canvas to the given dimensions.
-	//
-	// @param {number} width New width of the canvas, in pixels.
-	// @param {number} width New height of the canvas, in pixels.
+    angleCache = styleCache[angle];
+    if (angleCache == null) {
+      angleCache = styleCache[angle] = {};
+    }
 
-	Canvas.prototype.resize = function(width, height) {
+    info = angleCache[text];
 
-		if (width <= 0 || height <= 0) {
-			throw new Error("Invalid dimensions for plot, width = " + width + ", height = " + height);
-		}
+    // If we can't find a matching element in our cache, create a new one
 
-		var element = this.element,
-			context = this.context,
-			pixelRatio = this.pixelRatio;
+    if (info == null) {
+      const element = $('<div></div>')
+        .html(text)
```

Comment:
I don't understand why this was changed from the safer [`.text`](https://api.jquery.com/text/#text) method to [`.html`](https://api.jquery.com/html/)? Assuming the `text` argument really is text and not something else, I think we may want to change it back.

```suggestion
        .text(text)
```